### PR TITLE
User List Sorting Fix

### PR DESCRIPTION
**Pull request by prashant-tholia**
Fix user list sorting issue. Retain search term and sort order in URL query string on click on page navigation buttons.

Related issue: https://github.com/WeblateOrg/weblate/issues/4452

### CODE REVIEW

File: weblate/accounts/views.py [hunk 1]
```diff
@@ -1339,10 +1339,13 @@ def setup(self, request, *args, **kwargs):
             self.sort_query = "-date_joined"
 
     def get_context_data(self, **kwargs):
+        search = self.form.cleaned_data.get("q", "").strip()
+
```
Comment (nijel) [inline weblate/accounts/views.py:1343]: Please move this below the method comment. Or avoid the intermediate variable completely as it is used in single place only.

File: weblate/accounts/views.py [hunk 2]
```diff
@@ -1339,13 +1339,11 @@ def setup(self, request, *args, **kwargs):
...
-        search = self.form.cleaned_data.get("q", "").strip()
-
         """Create context for rendering page."""
         context = super().get_context_data(**kwargs)
         context["form"] = self.form
         context["sort_query"] = self.sort_query
         context["sort_name"] = self.form.sort_choices[self.sort_query.strip("-")]
         context["sort_choices"] = self.form.sort_choices
-        context["query_string"] = urlencode({"q": search, "sort_by": self.sort_query})
+        context["query_string"] = urlencode({"q": self.form.cleaned_data.get("q", "").strip(), "sort_by": self.sort_query})
```
Comment (nijel) [inline weblate/accounts/views.py]:
```suggestion
        context["query_string"] = urlencode({
            "q": self.form.cleaned_data.get("q", "").strip(), 
            "sort_by": self.sort_query,
        })
```

File: weblate/accounts/views.py [hunk 3]
```diff
@@ -1345,4 +1345,8 @@ def get_context_data(self, **kwargs):
         context["sort_query"] = self.sort_query
         context["sort_name"] = self.form.sort_choices[self.sort_query.strip("-")]
         context["sort_choices"] = self.form.sort_choices
+        context["query_string"] = urlencode({
+            "q": self.form.cleaned_data.get("q", "").strip(),
+            "sort_by": self.sort_query,
+        })
```
Comment (nijel) [inline weblate/accounts/views.py:1351]: Easist approach to deal with thisis to install pre-commit and let it do the code formatting, see https://docs.weblate.org/en/latest/contributing/code.html#coding-standard

This should be correct:

```suggestion
        context["query_string"] = urlencode(
            {
                "q": self.form.cleaned_data.get("q", "").strip(),
                "sort_by": self.sort_query,
            }
        )
```

Comment (prashant-tholia) [reply weblate/accounts/views.py]: Thanks @nijel . I've installed the packages as suggested at the link. I thank you for your patience.

**Comment by nijel** [comment]
Merged, thanks for your contribution!

**Comment by prashant-tholia** [comment]
@nijel, I see that the submission passed only 17 out of 26 checks. I request you to let me know what more was needed to be done so that it passed all checks. Thanks.

**Comment by nijel** [comment]
There were test failures in the master branch, so it was not related to this pull request. That's why I've ignored these failures and merged it.

**Comment by prashant-tholia** [comment]
Okay, thanks.

**Comment by prashant-tholia** [comment]
Hey, no worries. I'm thankful to you for being supportive.

**Pull request merged**